fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/provider/FuzzyQueryMatcher.groovy

```diff
@@ -0,0 +1,135 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+
+package com.yahoo.bard.webservice.data.config.provider
+
+import com.yahoo.bard.webservice.data.metric.TemplateDruidQuery
+import com.yahoo.bard.webservice.druid.model.aggregation.Aggregation
+import com.yahoo.bard.webservice.druid.model.postaggregation.ArithmeticPostAggregation
+import com.yahoo.bard.webservice.druid.model.postaggregation.ConstantPostAggregation
+import com.yahoo.bard.webservice.druid.model.postaggregation.FieldAccessorPostAggregation
+import com.yahoo.bard.webservice.druid.model.postaggregation.PostAggregation
+
+/**
+ * Utility class to validate that TemplateDruidQueries have the same structure and values.
+ *
+ * It's ok if inner objects don't always have the same names.
+ */
```

Comment:
It could be that there *shouldn't* be any, but the code I've written currently behaves that way (that code isn't in this PR). 

In the code I've written to parse/create logical metrics it creates temporary names for inner values (just increments a counter). Maybe there's a better way to do this, but it's simple/works.

So, something like `real_views = 2 * half_views + 7` would name the final logical metric `real_views` but would assign some temporary name to the inner nested metrics (e.g. `2*half_views`).